quipucords/tests/api/source/test_serializer.py

```diff
@@ -108,7 +108,7 @@ def test_openshift_source_default_port(openshift_cred_id):
     assert serializer.validated_data["port"] == 6443
 
 
-@pytest.mark.xpass("hosts ('5.4.3.2.1') is clearly invalid in this case")
+@pytest.mark.xfail(reason="hosts ('5.4.3.2.1') is clearly invalid in this case")
```

Comment:
I think the motivation for an "xpass" here (even though the marker does not work) is supposed to be broken, but it's not.

I'm not 100% sure if it should fail, tho. the assumption that it should fail comes from the premise that 5.4.3.2.1 was supposed to be an IP address BUT we do also accept URLs. Is this a valid URL? (I really don't know) If so, we can drop this mark. Otherwise, xfail is as wrong as the test is (since it should be testing for something invalid - OTH, if we really start a discussion on the merit of this negative test, it should actually be a for the IP address parsing function, not the serializer IMO.

TLDR: I don't even know this test should exist lol

Reply:
I think such test is fine overall (putting aside details like whether it should be focused on IP parsing function or serializer) - I strongly prefer most negative / unhappy path tests to be at unit level, which allows us to focus on actual functionality at higher levels (e.g. in Camayoc).

This does not seem to be a valid URL. URL may contain "host", which must be IPv6 (put inside square brackets), IPv4 (a sequence of four numbers separated by dot) or "registered name", as defined by [RFC 1034](https://datatracker.ietf.org/doc/html/rfc1034#section-3.5). According to that, last DNS component must start with a letter.

So something like "5.4.3.2.a1" could be a valid URL, but "5.4.3.2.1" is definitely not. It's not a valid IPv4 and it's not a valid registered name.

(I _think_ it's common for apps and libraries to keep a list of valid and recognized TLDs, so they would reject "5.4.3.2.a1" anyway; but it is a valid syntax.)

Reply:
The problem is `xpass` isn't even a valid marker. It's just apparently a random word we slapped on the test.

`xfail` means "this test _should_ fail but it's okay to expect it to pass" which I am pretty sure is the case for this bogus IP address.